token/cli/src/main.rs

```diff
@@ -2978,6 +3101,16 @@ fn app<'a, 'b>(
                         .takes_value(true)
                         .help("Expected fee amount collected during the transfer"),
                 )
+                .arg(
+                    Arg::with_name("transfer_hook_account")
+                        .long("transfer-hook-account")
+                        .validator(is_valid_pubkey)
+                        .value_name("PUBKEY")
+                        .takes_value(true)
+                        .multiple(true)
+                        .min_values(0u64)
+                        .help("Additional pubkey(s) required for a transfer hook. Used for offline transaction creation and signing.")
+                )
```

Comment:
In regards to your reply in #4548 , I'm much more in favor of the second suggestion:

```
--transfer-hook-account <PUBKEY>:readonlysigner
```

However, I'm also curious about using an array?

Reply:
Yeah that makes sense, I'm just lazy :sweat_smile: 

What do you mean about using an array?

Reply:
Does `clap` support - without too much clunkiness - a value that's an array? like:

```
--transfer-hook-accounts [<PUBKEY>:readonlysigner, <PUBKEY>:readonly, <PUBKEY>:readonly]
```

and then maybe optionally this can accept a path to a JSON or something

Reply:
By convention our tools use repeated args in case we want to provide it more than once. And personally, I find array arguments extremely unclear and clunky because there's no real convention. Typically I've seen it as:

```
--transfer-hook-accounts pubkey:type,pubkey:type,pubkey:type
```

So you split on the comma, which makes writing long arguments very ugly. To implement your suggestion, we'd need to write a specialized parser that looks at the `[` and `]` characters and goes through spaces, which is not typically how arg parsing is implemented.

Anyway, I'll add a new enum for the different meta combinations and parse it properly, ie:

```
enum AccountMetaRole {
    Readonly,
    Writable,
    ReadonlySigner,
    WritableSigner,
}
```

Reply:
Fair point(s)! Probably an anti-pattern and looks pretty ugly the more I stare at the comma-separated items.

If repeat args is the pattern then all good

Reply:
Done!